### PR TITLE
Migration of /services endpoint to `host_services` using `fastapi`

### PR DESCRIPTION
**Pull request by nissou91**

### CODE REVIEW

File: wazo_sysconfd/plugins/systemd_services/http.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2022 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from fastapi import APIRouter, Body
+from .services import services
+
+router = APIRouter()
+
+@router.post('/systemd_services', status_code=200)
```
Comment (sduthil) [inline wazo_sysconfd/plugins/systemd_services/http.py:9]: Please don't break the API in this refactoring branch: there are other services depending on the `/services` API (confd, setupd, maybe others). Please keep the API spec as it was. And yes, I totally agree that a `/services` API is a terrible name.

File: wazo_sysconfd/plugins/systemd_services/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+class InvalidActionException(ValueError):
+    def __init__(self, service_name, action):
+        super(InvalidActionException, self).__init__(self)
```
Comment (sduthil) [inline wazo_sysconfd/plugins/systemd_services/exceptions.py:7]: Python 3 allows the use of the shorter version `super().__init__()`

File: wazo_sysconfd/plugins/systemd_services/services.py [hunk 3]
```diff
@@ -66,12 +51,13 @@ def _run_action_for_service_validated(service, action):
         logger.debug("%s : return code %d", ' '.join(command), p.returncode)
 
         if p.returncode != 0:
-            raise HttpReqError(500, output)
+            raise exceptions.InternalServerErrorException(500, output)
```
Comment (sduthil) [inline wazo_sysconfd/plugins/systemd_services/services.py:54]: Changing the exception type will most likely break the error handling here. Please keep the original type.

File: wazo_sysconfd/plugins/systemd_services/services.py [hunk 4]
```diff
@@ -66,12 +51,13 @@ def _run_action_for_service_validated(service, action):
...
     except OSError:
         logger.exception("Error while executing action")
-        raise HttpReqError(500, "can't manage services")
-
-    return output
+        raise exceptions.InternalServerErrorException(500, "can't manage services")
```
Comment (sduthil) [inline wazo_sysconfd/plugins/systemd_services/services.py:57]: Idem

File: wazo_sysconfd/plugins/systemd_services/services.py [hunk 5]
```diff
@@ -66,12 +51,13 @@ def _run_action_for_service_validated(service, action):
...
 
+    if isinstance(output, bytes):
+        return output.decode("utf-8")
+    else:
+        return output
```
Comment (sduthil) [inline wazo_sysconfd/plugins/systemd_services/services.py:62]: You should use `Popen`'s `encoding` argument instead, like you did in https://github.com/wazo-platform/wazo-sysconfd/pull/48

**Review by sduthil — CHANGES_REQUESTED** [review]

File: wazo_sysconfd/plugins/systemd_services/services.py [hunk 6]
```diff
@@ -4,27 +4,12 @@
...
-
-
-class InvalidServiceException(ValueError):
-    def __init__(self, service_name):
-        super(InvalidServiceException, self).__init__(self)
-        self.service_name = service_name
-
-
-def services(args, options):
+def services(args):
```
Comment (sduthil) [inline wazo_sysconfd/plugins/systemd_services/services.py:12]: There are other modules depending on `wazo_sysconfd.modules.services`. Please update them accordingly.

**Review by sduthil — CHANGES_REQUESTED** [review]

**Comment by sduthil** [comment]
More generally, I disagree with the name `systemd_services`: `systemd` is only an implementation detail and should not appear in the module name. I would agree more with `system_services` or `host_services`. Other suggestions welcome :)

File: wazo_sysconfd/plugins/systemd_services/services.py (hunk 5 above)
Comment (nissou91) [reply wazo_sysconfd/plugins/systemd_services/services.py]: If I do like you suggested, the unit tests `test_services.py` will fail with the error `No Call`

File: wazo_sysconfd/plugins/systemd_services/services.py (hunk 6 above)
Comment (nissou91) [reply wazo_sysconfd/plugins/systemd_services/services.py]: Done, the only call I found is within `xivoctl` module

**Comment by nissou91** [comment]
> More generally, I disagree with the name `systemd_services`: `systemd` is only an implementation detail and should not appear in the module name. I would agree more with `system_services` or `host_services`. Other suggestions welcome :)

Thanks for the suggestions, I decided to go with `host_services`

File: wazo_sysconfd/plugins/systemd_services/services.py (hunk 5 above)
Comment (nissou91) [reply wazo_sysconfd/plugins/systemd_services/services.py]: Reverted

File: wazo_sysconfd/plugins/systemd_services/services.py (hunk 4 above)
Comment (nissou91) [reply wazo_sysconfd/plugins/systemd_services/services.py]: Reverted

File: wazo_sysconfd/plugins/systemd_services/services.py (hunk 3 above)
Comment (nissou91) [reply wazo_sysconfd/plugins/systemd_services/services.py]: Reverted

File: wazo_sysconfd/plugins/systemd_services/exceptions.py (hunk 2 above)
Comment (nissou91) [reply wazo_sysconfd/plugins/systemd_services/exceptions.py]: Done

File: wazo_sysconfd/plugins/systemd_services/http.py (hunk 1 above)
Comment (nissou91) [reply wazo_sysconfd/plugins/systemd_services/http.py]: Reverted

File: wazo_sysconfd/plugins/systemd_services/services.py (hunk 5 above)
Comment (sduthil) [reply wazo_sysconfd/plugins/systemd_services/services.py]: This error was already fixed in the `fastapi` branch.

**Review by sduthil — APPROVED** [review]

**Pull request merged**